### PR TITLE
templates: dynamic page rendering

### PR DESCRIPTION
**Pull request by lnielsen**
* Adds support for rendering pages as templates which allow to include e.g.
  configuration variables.

Signed-off-by: Lars Holm Nielsen <lars.holm.nielsen@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@lnielsen we should discuss this one as it allows server side code execution or also access to `SECRET_KEY` by any user that can edit the page.

Possible solutions:
- create new Jinja2 environment with selected filters and config variables
- use MD or RST
- replace dynamic parts with AJAX if need to be

**Comment by lnielsen** [comment]
> create new Jinja2 environment with selected filters and config variables

This is the only of the three options I see would work. However I think it's easier to simply take out the template from ``PAGES_TEMPLATES``, so that it's not enabled by default. Hence, if you enable it, you must know what you are doing.

File: invenio_pages/config.py [hunk 1]
```diff
@@ -29,5 +29,13 @@
...
 """Default template to render."""
 
-PAGES_TEMPLATES = [('invenio_pages/default.html', 'Default'), ]
+PAGES_TEMPLATES = [
+    ('invenio_pages/default.html', 'Default'),
+    ('invenio_pages/dynamic.html', 'Default dynamic'),
+]
 """List of available templates for pages."""
+
+PAGES_CONFIGVAR_WHITELIST = [
```
Comment (jirikuncar) [inline invenio_pages/config.py]: What would you say about `PAGES_WHITELIST_CONFIG_KEYS`?

File: invenio_pages/templates/invenio_pages/dynamic.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{#
+# This file is part of Invenio.
+# Copyright (C) 2014, 2015 CERN.
```
Comment (jirikuncar) [inline invenio_pages/templates/invenio_pages/dynamic.html]: `2016`?

**Comment by jirikuncar** [comment]
- [ ] invenio_pages/views.py add 4:`, 2016 `

File: tests/test_views.py [hunk 3]
```diff
@@ -49,6 +49,32 @@ def test_page_content(pages_fixture):
...
+    app.config['PAGES_WHITELIST_CONFIG_KEYS'] = ['MYVAR']
+    InvenioPages(app)
+    app.register_blueprint(blueprint)
+    content = 'dynamic-content'
+    with app.app_context():
+        app.config['MYVAR'] = content
+        page = Page(
+            url='/dynamic',
+            title='Dynamic page',
+            content='{{MYVAR}}',
```
Comment (jirikuncar) [inline tests/test_views.py]: shall we check that one doesn't see `current_app`, `config` or `SECRET_KEY`?

Comment (lnielsen) [inline tests/test_views.py]: I've added an issue: #23

Comment (jirikuncar) [inline tests/test_views.py]: :+1:

**Comment by jirikuncar** [comment]
@lnielsen `3: M190 line is too long (75 > 72)`

**Comment by lnielsen** [comment]
Where?

**Comment by jirikuncar** [comment]
@lnielsen in commit message (`kwalitee check message`).

**Comment by lnielsen** [comment]
Updated. Funny, always thought it was 79 like Python but I can see its been 72 all along. We really need urgently to make kwalitee part of run-tests.sh and Travis jobs.

**Comment by tiborsimko** [comment]
@lnielsen Even better, you can plug a git commit message mode to your editor, so that the line length and whatnot will be OK from the start, even before kwalitee sees it.  Both Emacs and Vim has such a mode; not sure about Sublime git integration.

**Pull request merged**